ledger-core/virtual/integration/deduplication_test.go

```diff
@@ -318,3 +321,525 @@ func TestDeduplication_SecondCallOfMethodAfterExecution(t *testing.T) {
+					State:     []byte("object memory"),
+				},
+			},
+		}
+
+		if testInfo.pending {
+			report.OrderedPendingCount = 1
+			report.OrderedPendingEarliestPulse = s.getP1()
+		}
+
+		s.server.SendPayload(ctx, &report)
+
+		return false // no resend msg
+	}).ExpectedCount(1)
+
+	if testInfo.expectFindRequestMessage {
+		typedChecker.VFindCallRequest.Set(func(req *payload.VFindCallRequest) bool {
+			require.Equal(t, s.getP1(), req.LookAt)
+			require.Equal(t, s.getObject(), req.Callee)
+			require.Equal(t, s.getOutgoingRef(), req.Outgoing)
+
+			response := payload.VFindCallResponse{
+				Callee:   s.getObject(),
+				Outgoing: s.getOutgoingRef(),
+				Status:   testInfo.findRequestStatus,
+			}
+
+			if testInfo.findRequestHasResult {
+				response.CallResult = &payload.VCallResult{
+					CallType:        payload.CTMethod,
+					CallFlags:       payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+					Caller:          s.getCaller(),
+					Callee:          s.getObject(),
+					CallOutgoing:    s.getOutgoingLocal(),
+					ReturnArguments: []byte("found request"),
+				}
+			}
+
+			s.server.SendPayload(ctx, &response)
+			return false
+		}).ExpectedCount(1)
+	}
+
+	if testInfo.expectResultMessage {
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			require.Equal(t, payload.CTMethod, res.CallType)
+			flags := payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty)
+			require.Equal(t, flags, res.CallFlags)
+			require.Equal(t, s.getCaller(), res.Caller)
+			require.Equal(t, s.getObject(), res.Callee)
+			require.Equal(t, s.getOutgoingLocal(), res.CallOutgoing)
+
+			if testInfo.expectExecution {
+				require.Equal(t, []byte("execution"), res.ReturnArguments)
+			} else {
+				require.Equal(t, []byte("found request"), res.ReturnArguments)
+			}
+
+			return false
+		}).ExpectedCount(1)
+	}
+}
+
+func (s *deduplicateMethodUsingPrevVETest) setRunnerMock() {
+	isolation := contract.MethodIsolation{Interference: contract.CallTolerable, State: contract.CallDirty}
+	s.runnerMock.AddExecutionClassify("SomeMethod", isolation, nil)
+
+	newObjDescriptor := descriptor.NewObject(
+		reference.Global{}, reference.Local{}, s.getClass(), []byte(""), reference.Global{},
+	)
+
+	requestResult := requestresult.New([]byte("execution"), gen.UniqueGlobalRef())
+	requestResult.SetAmend(newObjDescriptor, []byte("new memory"))
+
+	executionMock := s.runnerMock.AddExecutionMock("SomeMethod")
+	executionMock.AddStart(func(ctx execution.Context) {
+		s.incNumberOfExecutions()
+	}, &execution.Update{
+		Type:   execution.Done,
+		Result: requestResult,
+	})
+}
+
+func (s *deduplicateMethodUsingPrevVETest) waitMessagePublications(
+	ctx context.Context,
+	t *testing.T,
+	expected int,
+) {
+	t.Helper()
+	if !s.server.PublisherMock.WaitCount(expected, 10*time.Second) {
+		require.Fail(t, "timeout waiting for messages on publisher")
+	}
+	require.Equal(t, expected, s.server.PublisherMock.GetCount())
```

Comment:
You can use it
```
	require.Equal(t, 4, typedChecker.VCallRequest.Count())
	require.Equal(t, 2, typedChecker.VCallResult.Count())
```

Reply:
done

Reply:
function `waitMessagePublications` can be removed now, am I right?

Reply:
yep, thanks, done